### PR TITLE
[browser-settings] Add search engine model. JB#40380

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

File: apps/browser/settings/searchenginemodel.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        return QVariant();
+    }
+}
+
+QHash<int, QByteArray> SearchEngineModel::roleNames() const
+{
+    QHash<int, QByteArray> roles;
+    roles[UrlRole] = "url";
+    roles[TitleRole] = "title";
+    roles[InstalledRole] = "date";
```
Comment (rainemak) [inline apps/browser/settings/searchenginemodel.cpp:58]: Typo... role must be "installed" of course.

Comment (rainemak) [reply apps/browser/settings/searchenginemodel.cpp]: Fixed

**Review by mkenttala — APPROVED** [review]

File: apps/browser/settings/searchenginemodel.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            : url(url)
+            , title(title)
+            , installed(installed)
+        {}
+
+        QUrl url;
+        QString title;
+        bool installed;
+    };
+
```
Comment (rainemak) [inline apps/browser/settings/searchenginemodel.h:60]: Extra empty line. Will remove this one.

File: apps/browser/settings/searchenginemodel.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    QList<SearchEngine> m_searchEngines;
+
```
Comment (rainemak) [inline apps/browser/settings/searchenginemodel.h:63]: Extra empty line. Will remove this one.

**Pull request merged**